examples/notebooks/recsys2021/RecsysSchema.py

```diff
@@ -67,39 +67,39 @@ def __init__(self):
     def toStructType(self):
         str_fields1 = [StructField('%s' % i, StringType())
                        for i in self.string_cols1]
-        int_fields1 = [StructField('%s' % i, IntegerType())
-                       for i in self.int_cols1]
+        long_fields1 = [StructField('%s' % i, LongType())
+                       for i in self.long_cols1]
         str_fields2 = [StructField('%s' % i, StringType())
                        for i in self.string_cols2]
-        int_fields2 = [StructField('%s' % i, IntegerType())
-                       for i in self.int_cols2]
+        long_fields2 = [StructField('%s' % i, LongType())
+                       for i in self.long_cols2]
         bool_fields1 = [StructField('%s' % i, BooleanType())
                         for i in self.bool_cols1]
-        int_fields3 = [StructField('%s' % i, IntegerType())
-                       for i in self.int_cols3]
+        long_fields3 = [StructField('%s' % i, LongType())
+                       for i in self.long_cols3]
         str_fields3 = [StructField('%s' % i, StringType())
                        for i in self.string_cols3]
-        int_fields4 = [StructField('%s' % i, IntegerType())
-                       for i in self.int_cols4]
+        long_fields4 = [StructField('%s' % i, LongType())
+                       for i in self.long_cols4]
         bool_fields2 = [StructField('%s' % i, BooleanType())
                         for i in self.bool_cols2]
-        int_fields5 = [StructField('%s' % i, IntegerType())
-                       for i in self.int_cols5]
+        long_fields5 = [StructField('%s' % i, LongType())
+                       for i in self.long_cols5]
         bool_fields3 = [StructField('%s' % i, BooleanType())
                         for i in self.bool_cols3]
-        float_fields = [StructField('%s' % i, FloatType())
-                        for i in self.float_cols]
+        double_fields = [StructField('%s' % i, DoubleType())
+                        for i in self.double_cols]
         return StructType(
             str_fields1
-            + int_fields1
+            + long_fields1
             + str_fields2
-            + int_fields2
+            + long_fields2
             + bool_fields1
-            + int_fields3
+            + long_fields3
             + str_fields3
-            + int_fields4
+            + long_fields4
             + bool_fields2
-            + int_fields5
+            + long_fields5
             + bool_fields3
-            + float_fields
+            + double_fields
```

Comment:
@XinyaoWa , the reason of changing this is ?

Reply:
To make the schema align with the training data input format.

Reply:
I see, but I think this should be reversed, we should convert feature to int/float before train, instead of make them as long/double. I assume the original idea Bin did here is to make data memory capacity smaller

Reply:
OK, I have modified it.